schema.ts

```diff
@@ -1232,6 +1232,82 @@ export interface ObjectSchemaDefinition<K extends string, L extends string>
   // TODO(dweitzman): Only support options in the typing when the codaType is ValueHintType.SelectList.
 }
 
+/**
+ * Discriminent used to determine the permission type
+ * @hidden
+ */
+export enum PrincipalType {
+  User = 'user',
+  Group = 'group',
+  Domain = 'domain',
+  Public = 'public',
+}
+
+/**
+ * This type definition represents properties required for all permissions.
+ *
+ * TODO(sam): Unhide this
+ * @hidden
+ */
+interface BasePermission {
+  type: PrincipalType;
+  rowId: string | number;
```

Comment:
cc @ekoleda-codaio since this is also deviating from the pattern of executeUpdate's return object structure.

Reply:
This has changed a bit based on @jonathan-codaio 's comments but I thought this was a bit more explicit then just row ordering